src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -274,19 +280,48 @@ internal CompilationUnitSyntax ParseCompilationUnitCore()
                 openBrace = this.ConvertToMissingWithTrailingTrivia(openBrace, SyntaxKind.OpenBraceToken);
             }
 
+            Debug.Assert(semicolon != null || openBrace != null);
+
             var body = new NamespaceBodyBuilder(_pool);
-            SyntaxListBuilder initialBadNodes = null;
             try
             {
-                this.ParseNamespaceBody(ref openBrace, ref body, ref initialBadNodes, SyntaxKind.NamespaceDeclaration);
+                if (openBrace == null)
+                {
+                    Debug.Assert(semicolon != null);
 
-                var closeBrace = this.EatToken(SyntaxKind.CloseBraceToken);
-                var semicolon = this.TryEatToken(SyntaxKind.SemicolonToken);
+                    SyntaxListBuilder initialBadNodes = null;
+                    this.ParseNamespaceBody(ref semicolon, ref body, ref initialBadNodes, SyntaxKind.SingleLineNamespaceDeclaration);
+                    Debug.Assert(initialBadNodes == null); // init bad nodes should have been attached to open brace...
```

Comment:
Comment is wrong (would be attached to semicolon). Do we have a test scenario where that happens?

Reply:
@RikkiGibson to address

Reply:
See `SingleLineDeclarationParsingTests.NamespaceWithExtraSemicolon`. Will fix the comment.